sphinxcontrib/openapi/openapi30.py

```diff
@@ -264,22 +309,46 @@ def _httpresource(endpoint, method, properties, convert, render_examples,
             yield '{indent}{indent}{line}'.format(**locals())
         if param.get('required', False):
             yield '{indent}{indent}(Required)'.format(**locals())
+    # security params
+    securities = properties.get('security', [])
+    for security in securities:
+        for ref in security.keys():
+            secScheme = components.get('securitySchemes', {}).get(ref, {})
+            if secScheme['type'] == 'apiKey':
+                if secScheme['in'] == 'query':
+                    yield indent + ':query {type} {name}:'.format(
+                        type='API key',
+                        name=secScheme['name'])
+                    yield ('{indent}{indent}(Required'
+                           ' by authentication "{ref}")').format(**locals())
 
     # print request content
     if render_request:
+        yield ''
         request_content = properties.get('requestBody', {}).get('content', {})
         if request_content and 'application/json' in request_content:
             schema = request_content['application/json']['schema']
-            req_properties = json.dumps(schema['properties'], indent=2,
-                                        separators=(',', ':'))
-            yield '{indent}**Request body:**'.format(**locals())
-            yield ''
-            yield '{indent}.. sourcecode:: json'.format(**locals())
-            yield ''
-            for line in req_properties.splitlines():
-                # yield indent + line
-                yield '{indent}{indent}{line}'.format(**locals())
-                # yield ''
+            schs = []
+            if 'oneOf' in schema:
```

Comment:
Hm.. I wonder, is it a good idea to render every possible body input? :thinking: Can you please add tests?

Reply:
Sorry, just seeing your reviews now, I'll address them ASAP!